model/perf_data.go

```diff
@@ -36,18 +50,171 @@ type PerfRollups struct {
 
 	dirty     bool
 	populated bool
+	id        string
 	env       sink.Environment
 }
 
-func (r *PerfRollups) Setup(env sink.Environment)                            { r.env = env }
-func (r *PerfRollups) Add(name string, version int, value interface{}) error {}
-func (r *PerfRollups) GetInt(name string) (int, error)                       {}
-func (r *PerfRollups) GetInt32(name string) (int32, error)                   {}
-func (r *PerfRollups) GetInt64(name string) (int64, error)                   {}
-func (r *PerfRollups) GetFloat(name string) (float64, error)                 {}
-func (r *PerfRollups) Validate() error                                       {}
-func (r *PerfRollups) Map() map[string]int64                                 {}
-func (r *PerfRollups) MapFloat() map[string]float64                          {}
+type perfRollupEntries struct {
```

Comment:
I think this might be better as a type alias rather than a struct....

Reply:
I think this needs to be a struct because One will not marshal into a list.

Reply:
I'd just do something with anon structs for the purpose of unmarshalling. you can also just use the larger model and the decoder will just leave the fields empty.